libgnucash/engine/test/test-qofbook.c

```diff
@@ -768,6 +769,25 @@ test_book_get_collection( Fixture *fixture, gconstpointer pData )
     g_assert( m_col == m_col2 );
 }
 
+
+static void
+test_book_features (Fixture *fixture, gconstpointer pData)
+{
+    g_test_message ("Testing book features");
+
+    g_assert_null (gnc_features_test_unknown (fixture->book));
+    g_assert_false (gnc_features_check_used (fixture->book, "Credit Notes"));
+
+    gnc_features_set_used (fixture->book, "Credit Notes");
+    g_assert_null (gnc_features_test_unknown (fixture->book));
+    g_assert_true (gnc_features_check_used (fixture->book, "Credit Notes"));
+
+    /* cannot set an unknown feature: it bails out. */
+    /* gnc_features_set_used (fixture->book, "Quantum Computing and Nanotech"); */
+    /* g_assert_nonnull (gnc_features_test_unknown (fixture->book)); */
+    g_assert_false (gnc_features_check_used (fixture->book, "Quantum Computing and Nanotech"));
```

Comment:
I don't know how to feed a datafile with unknown features without using the gnc-features.h functions.

Reply:
I'm not sure I understand the question. If "feed" means "load", you can't. That's the whole point of the features check, to prevent GnuCash from opening a file created with features it doesn't know about because it might do something that would break the part of the file that depends on that unknown feature.

If you mean something else, please expand on it.

Reply:
That's exactly it. It's for the tests only. The `g_assert_nonnull (gnc_features_test_unknown (fixture->book))` cannot be tested in tests.